### PR TITLE
fix mailru/easyjson#310

### PR DESCRIPTION
**Pull request by YaroslavPodorvanov**

### CODE REVIEW

**Comment by YaroslavPodorvanov** [comment]
@kirillx can you approve this?

File: jlexer/lexer.go [hunk 1]
```diff
@@ -667,6 +667,8 @@ func (r *Lexer) String() string {
 	}
 	var ret string
 	if r.token.byteValueCloned {
+		r.token.byteValueCloned = false
```
Comment (pkarrot) [inline jlexer/lexer.go:670]: Any details on why it fixes the issue?

Actually better to provide details why the issue occurs at all... byteValueCloned should be TRUE only iff the value was allocated and no longer refers to original buffer. So this TRUE path should be safe. Why it turns out it is not?

Comment (kirillx) [inline jlexer/lexer.go:670]: Any details on why it fixes the issue? cloned values are immutable and should be safe to return... what is really wrong here?

Actually better to provide details why the issue occurs at all... byteValueCloned should be TRUE only iff the value was allocated and no longer refers to original buffer. So this TRUE path should be safe. Why it turns out it is not?

Comment (YaroslavPodorvanov) [reply jlexer/lexer.go:670]: After first optimization *byteValueCloned* stays always *true* and in next property can clone *mutable* value like in #310

Comment (kirillx) [reply jlexer/lexer.go:670]: I think the nicer looking fix is smth like that instead:

```
--- a/jlexer/lexer.go
+++ b/jlexer/lexer.go
@@ -401,6 +401,7 @@ func (r *Lexer) scanToken() {
 // consume resets the current token to allow scanning the next one.
 func (r *Lexer) consume() {
        r.token.kind = tokenUndef
+       r.token.byteValueCloned = false
        r.token.delimValue = 0
 }
```

it guarantees that lexer state is correctly reset in a single place.

Comment (YaroslavPodorvanov) [reply jlexer/lexer.go]: @kirillx great, did move *r.token.byteValueCloned = false* as your example

**Review by kirillx — APPROVED** [review]

**Comment by kirillx** [comment]
@GoWebProd please take a look. Fix looks obvious and I checked it manually on test case similar to the one in bug.
If Yaroslav doesn't add test case in his PR I can do it separately.

**Comment by GoWebProd** [comment]
Hello, thanks.

**Comment by sobolevsv** [comment]
when will it be released?

**Pull request merged**